spec/re2/set_spec.rb

```diff
@@ -0,0 +1,96 @@
+RSpec.describe RE2::Set do
+  # libre2, as of 2022-06-01, in set.cc, does print to stderr (fd 2) even when
+  # :log_errors is true. So we need to temporarily redirect stderr while running
+  # tests that might print to it. The method used here is taken from the source
+  # code of MRI and seems to do the job, but it isn't guaranteed to cover all
+  # cases.
+
+  NullDevice = defined?(IO::NULL) ? IO::NULL :
+    %w[/dev/null NUL NIL: NL:].find { |dev| File.exist?(dev) }
+
+  def silence_streams(*streams)
+    on_hold = streams.collect { |stream| stream.dup }
+    streams.each do |stream|
+      stream.reopen(NullDevice)
+      stream.sync = true
+    end
+    yield
+  ensure
+    streams.each_with_index do |stream, i|
+      stream.reopen(on_hold[i])
+    end
+  end
```

Comment:
Rather than reopening `STDERR` to a platform-specific null device, could we take [a leaf from RSpec](https://github.com/rspec/rspec-expectations/blob/43bf64b01f8356979ffbc373b2e81d2ab1389b29/lib/rspec/matchers/built_in/output.rb#L159-L176) and re-assign `$stderr` to a disposable `StringIO` we let Ruby garbage collect without ever reading? e.g.

```ruby
def silence_stderr(&blk)
  null_stream = StringIO.new
  original_stream = $stderr
  $stderr = null_stream

  blk.call
ensure
  $stderr = original_stream
end

# ...

silence_stderr do
  expect { set.add("def") }.to raise_error(ArgumentError)
end
```

Reply:
I wasn't sure if simply reassigning a ruby variable would also change any libc-level `stderr` global variable, so I went looking at how MRI itself does this. I've now tested this snippet from RSpec and verified it works, so will change it.

Reply:
You were right about this: I switched back to reopening `STDERR` with a platform-specific null device in ecbeae6c00fdf912fe993924253c981496fd3d2e.

Reply:
You were right about this: I've reverted back to reopening `STDERR` with the platform-specific null device in ecbeae6c00fdf912fe993924253c981496fd3d2e.